packages/waku/src/lib/plugins/vite-plugin-rsc-delegate.ts

```diff
@@ -36,10 +38,17 @@ export function rscDelegatePlugin(
               const source = base + '@id/__x00__' + item.source.value;
               importCallback(source);
             } else if (CSS_LANGS_RE.test(item.source.value)) {
-              const filePath = path.join(path.dirname(id), item.source.value);
-              // HACK this relies on Vite's internal implementation detail.
-              const source = base + '@fs' + encodeFilePathToAbsolute(filePath);
-              importCallback(source);
+              const resolvedSource = await server.pluginContainer.resolveId(
+                item.source.value,
+                id,
+                { ssr: true },
+              );
+              if (resolvedSource?.id) {
+                const transformedResult = await server.transformRequest(
+                  resolvedSource.id,
+                );
+                transformedResult && importCallback(transformedResult);
+              }
```

Comment:
I do not fully understand this, but if works, it would be fine. I wonder if @himself65 has some experience.

Reply:
`pluginContainer` can run the vite to a specific file

Reply:
First, the previous implementation always assumes there's a filesystem import, which is wrong. 

Two, resolveId would run all of the plugins' resolveId function to resolve the right path for the import (whether local or npm module). When we get the right path for that, we transform the request so we get a vite-compatible module result. For instance, if we transformRequest a CSS file, in this case, vite would give us a hmr js module code specific to that CSS file, so when we use that code, we enable CSS in hmr and it'd also be linked to the document (CSS would get applied) 

Let me know if there's any thing vague.

Reply:
thanks!